src/hooks/mongodb.js

```diff
@@ -0,0 +1,86 @@
+import { safeRequire } from '../utils/requireUtils';
+import { getRandomId, safeExecute } from '../utils';
+import { createMongoDbSpan, extendMondoDbSpan } from '../spans/mongoDbSpan';
+import { SpansContainer } from '../globals';
+import * as logger from '../logger';
+
+const PendingRequests = (() => {
+  let pendingRequests = {};
+
+  const addPendingRequest = (requestId, spanId) => {
+    pendingRequests[requestId] = spanId;
+  };
+
+  const getPendingRequestSpanId = requestId => {
+    const spanId = pendingRequests[requestId];
+    delete pendingRequests[requestId];
```

Comment:
This function isn't pure and it's not clear from the name of the function.
Maybe add `delete` in the name of the function.

Reply:
pop